### PR TITLE
[Parser] Support rename this's property for IIFE

### PR DESCRIPTION
**Pull request by ljqx**
**What kind of change does this PR introduce?**
bugfix

**Did you add tests for your changes?**
Yes

**If relevant, link to documentation update:**
N/A

**Summary**

This is for fixing #5067.
ProvidePlugin supports replacing IIFE argument's property with resolved modules, but not `this`'s. This PR makes it available.

For example, for code
```
(function(){
  window.export = this.jQuery;

}).call(window);
```
config 
```
new webpack.ProvidePlugin({
  'window.jQuery': 'jquery',
})
```

will replace the `jQuery` with resolved one.

**Does this PR introduce a breaking change?**
No

**Other information**
This PR only handles `iife.call()` or `iife.bind()` style call of IIFE, not 
1. The simple `iife()` style as it has more complicated `this` resolving rules (environment, strict mode or not).
2. `.apply`
3. arrow functions
3. object method
4. getter/setter
5. constructor (seems not needed)
6. callback

### CODE REVIEW

File: lib/Parser.js [hunk 1]
```diff
@@ -1059,16 +1059,18 @@ class Parser extends Tapable {
...
+					if(!this.applyPluginsBailResult1("rename " + renameIdentifier, argOrThis))
 						return renameIdentifier;
 				}
-				this.walkExpression(arg);
-			}, this);
+				this.walkExpression(argOrThis);
+			}
+			const params = functionExpression.params;
+			const args = options.map(renameArgOrThis, this);
+			const renameThis = currentThis ? renameArgOrThis.call(this, currentThis) : null;
```
Comment (sokra) [inline lib/Parser.js]: Could you switch this line with the previous line? So that arguments are evaluated in the correct order.

Comment (ljqx) [inline lib/Parser.js]: sure

**Comment by ljqx** [comment]
Hi, @sokra , ci/circleci failed because some unrelated tests timeout. How could I force reevaluate the tests? thanks.

**Comment by ljqx** [comment]
Hi, @sokra , the comments are resolved, ci checks are passed too. Could you help continue the review? Thanks.

**Comment by sokra** [comment]
Thanks

**Pull request merged**